SCons/Script/Main.py

```diff
@@ -517,6 +519,26 @@ def GetOption(name):
 def SetOption(name, value):
     return OptionsParser.values.set_option(name, value)
 
+def DebugOptions(json=None):
+    """
+    API to allow specifying options to SCons debug logic
+    Currently only json is supported which changes the
+    json file written by --debug=json from the default
+    """
+    if json is not None:
+        json_node = SCons.Defaults.DefaultEnvironment().arg2nodes(json)
+        SCons.Util.stats.JSON_OUTPUT_FILE = json_node[0].get_abspath()
+        # Check if parent dir to JSON_OUTPUT_FILE exists
+        json_dir = os.path.dirname(SCons.Util.stats.JSON_OUTPUT_FILE)
+        try:
+            if not os.path.isdir(json_dir):
+                os.makedirs(json_dir, exist_ok=True)
+            # Now try to open file and see if you can..
+            open(SCons.Util.stats.JSON_OUTPUT_FILE,'w')
```

Comment:
Do you expect this to remain open from here? Or is this just a quick test-open? It's opened "for real" in `Util`, right?  Suggest using a `with` block here for clean resource allocation.

Reply:
yeah, we have other "test opens" - dblite does that for example.